syncer/checkpoint.go

```diff
@@ -740,7 +740,11 @@ func (cp *RemoteCheckPoint) genUpdateSQL(cpSchema, cpTable string, location binl
 	if len(tiBytes) == 0 {
 		tiBytes = []byte("null")
 	}
-	args := []interface{}{cp.id, cpSchema, cpTable, location.Position.Name, location.Position.Pos, location.GTIDSet.String(), tiBytes, isGlobal}
+	gsetStr := ""
+	if location.GTIDSet != nil {
+		gsetStr = location.GTIDSet.String()
+	}
```

Comment:
maybe a helper function can be added too.

Reply:
add in c5bfffaa928f9a81094faa6a37216a79279d65f0